CONTRIBUTING.md

```diff
@@ -4,6 +4,24 @@ This document should concisely express the project development status,
 methodology, and contribution process.  As the community makes progress, we
 should keep this document in sync with reality.
 
+## Submitting a Pull Request (PR)
+
+The following outlines the general rules we follow:
+
+- All PRs must have the appropriate documentation changes made within the
+same PR. Note, not all PRs will necessarily require a documentation change
+but if it does please include it in the same PR so the PR is complete and
+not just a partial solution.
+- All PRs must have the appropriate testcases. For example, bug fixes should
```

Comment:
Agree, but the verbiage here, if I put my son-of-two-lawyers hat on, sets a very high bar.  Normally a healthy codebase does not have 100% coverage.  There are also likely to be certain tests that are prohibitively complex to right in terms of the effort to write versus payoff ratio.  Please add some language around 'within reason' so it's clear that perfection is not the enemy of progress.